tests/rados/test_rados_watcher_cleanup.py

```diff
@@ -0,0 +1,825 @@
+    device: str,  # Device path on client like /dev/rbd0
+    rados_orchestrator_obj: RadosOrchestrator,
+    pool: str,
+    client_ip: str,
+) -> None:
+    """
+    Deletes an RBD watcher using a specified method and verifies its removal.
+
+    Args:
+        method (str): The cleanup method to apply (e.g., "unmap", "blacklist", "restart", "crash", "umount").
+            Example: "unmap"
+        client (Node): The client node object where the watcher resides.
+            Example: ceph_cluster.get_nodes(role="client")[0]
+        image_name (str): The name of the RBD image.
+            Example: "img_single_client_test_0"
+        device (str): The device path (e.g., /dev/rbd0) if applicable for unmap/umount.
+            Example: "/dev/rbd0"
+        rados_orchestrator_obj (RadosOrchestrator): The RadosOrchestrator object for
+                                                    executing Ceph commands and getting watcher info.
+            Example: rados_obj (instance created in run function)
+        pool (str): The name of the pool.
+            Example: "watcher_test_pool"
+        client_ip (str): The IP address of the client whose watcher is being deleted.
+            Example: "192.168.1.100"
+
+    Returns:
+        None
+    """
+    log.info(
+        "Attempting to delete watcher using '%s' for image: %s on client %s",
+        method,
+        image_name,
+        client.hostname,
+    )
+
+    mount_path: str = f"/mnt/{image_name}"
+
+    if method == "unmap":
+        log.info("Performing rbd unmap of device %s.", device)
+        out, err = client.exec_command(
+            cmd=f"rbd unmap {device}", sudo=True, check_ec=False
+        )
+        if err:
+            log.warning("rbd unmap stderr: %s", err.strip())
+        log.debug("rbd unmap stdout: %s", out.strip())
+        time.sleep(5)  # Give time for unmap to propagate
+        # After unmap, the device is no longer associated. Clean up mount point.
+        client.exec_command(cmd=f"rm -rf {mount_path}", sudo=True)
+        log.info("Cleaned up mount point %s after unmap.", mount_path)
+
+    elif method == "umount":
+        log.info("Attempting to unmount %s from mount path %s.", device, mount_path)
+        # Verify if it's actually mounted before attempting umount
+        mount_check_out, _ = client.exec_command(
+            cmd=f"findmnt {mount_path}", sudo=True, check_ec=False
+        )
+        if mount_path not in mount_check_out:
+            log.warning(
+                "Mount path %s not found to be mounted. Skipping umount for %s.",
+                mount_path,
+                image_name,
+            )
+        else:
+            out, err = client.exec_command(
+                cmd=f"umount {mount_path}", sudo=True, check_ec=False
+            )
+            if err:
+                log.warning("umount stderr: %s", err.strip())
+            log.debug("umount stdout: %s", out.strip())
+            time.sleep(5)  # Give time for umount to propagate
```

Comment:
Please validate umount command output for verification purposes, this should help eliminate the if-else conditional check
```
[root@folio01 ubuntu]# umount /nvn/dfd
umount: /nvn/dfd: no mount point specified.
[root@folio01 ubuntu]# umount /mnt/dfd
umount: /mnt/dfd: no mount point specified.
[root@folio01 ubuntu]# echo $?
32
```